source/common/common/safe_memcpy.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <cstring>
+
+namespace Envoy {
+
+/**
+ * @brief Copies src to dst based on their sizes, which must be the same.
```

Comment:
nit: we don't use `\brief` keyword in Envoy, but we do describe parameters, so something more like:
```
/**
 * Copies src to dst based on their sizes, which must be the same.
 * @param dst the destination
 * @param src the source
 */
```
here and below.